src/NuGet.Services.SearchService/App_Start/WebApiConfig.cs

```diff
@@ -157,14 +150,109 @@ private static AutofacWebApiDependencyResolver GetDependencyResolver(HttpConfigu
             builder.AddAzureSearch();
 
             var loggerConfiguration = LoggingSetup.CreateDefaultLoggerConfiguration(withConsoleLogger: false);
-            var loggerFactory = LoggingSetup.CreateLoggerFactory(loggerConfiguration);
+            var loggerFactory = LoggingSetup.CreateLoggerFactory(
+                loggerConfiguration,
+                telemetryConfiguration: applicationInsightsConfiguration.TelemetryConfiguration);
+
             builder.RegisterInstance(loggerFactory).As<ILoggerFactory>();
             builder.RegisterGeneric(typeof(Logger<>)).As(typeof(ILogger<>));
 
             var container = builder.Build();
             return new AutofacWebApiDependencyResolver(container);
         }
 
+        private static ApplicationInsightsConfiguration InitializeApplicationInsights(RefreshableConfiguration configuration)
+        {
+            var instrumentationKey = configuration.Root.GetValue<string>("ApplicationInsights_InstrumentationKey");
+            var heartbeatIntervalSeconds = configuration.Root.GetValue("ApplicationInsights_HeartbeatIntervalSeconds", 60);
+
+            var applicationInsightsConfiguration = ApplicationInsights.Initialize(
+                    instrumentationKey,
+                    TimeSpan.FromSeconds(heartbeatIntervalSeconds));
+
+            applicationInsightsConfiguration.TelemetryConfiguration.TelemetryInitializers.Add(new AzureWebAppTelemetryInitializer());
+
+            RegisterApplicationInsightsTelemetryModules(applicationInsightsConfiguration.TelemetryConfiguration);
+
+            return applicationInsightsConfiguration;
+        }
+
+        private static void RegisterApplicationInsightsTelemetryModules(TelemetryConfiguration configuration)
+        {
+            RegisterApplicationInsightsTelemetryModule(
+                new Microsoft.ApplicationInsights.WindowsServer.AppServicesHeartbeatTelemetryModule(),
+                configuration);
+
+            RegisterApplicationInsightsTelemetryModule(
+                new Microsoft.ApplicationInsights.WindowsServer.AzureInstanceMetadataTelemetryModule(),
```

Comment:
Why not a using?

Reply:
It was convenient when migrating from `applicationinsights.config` file, but other than that no particular reason.

Addressed with https://github.com/NuGet/NuGet.Services.Metadata/pull/732/commits/5d8b42da043cc692527046eb96609c4959df56ad